From akpm@osdl.org Thu May 11 16:09:29 2006
Return-Path: <akpm@osdl.org>
X-Original-To: kernel@kolivas.org
Delivered-To: kernel@kolivas.org
Received: from bhhdoa.org.au (bhhdoa.org.au [65.98.99.88])
	by mail.kolivas.org (Postfix) with ESMTP id CB51FC5CF4
	for <kernel@kolivas.org>; Thu, 11 May 2006 16:12:18 +1000 (EST)
Received: from smtp.osdl.org (smtp.osdl.org [65.172.181.4])
	by bhhdoa.org.au (Postfix) with ESMTP id 5ECD0516E6
	for <kernel@kolivas.org>; Thu, 11 May 2006 14:06:35 +1000 (EST)
Received: from shell0.pdx.osdl.net (fw.osdl.org [65.172.181.6])
	by smtp.osdl.org (8.12.8/8.12.8) with ESMTP id k4B6CFtH002262
	(version=TLSv1/SSLv3 cipher=EDH-RSA-DES-CBC3-SHA bits=168 verify=NO);
	Wed, 10 May 2006 23:12:16 -0700
Received: from localhost.localdomain (shell0.pdx.osdl.net [10.9.0.31])
	by shell0.pdx.osdl.net (8.13.1/8.11.6) with ESMTP id k4B6CFPG005351;
	Wed, 10 May 2006 23:12:15 -0700
Message-Id: <200605110612.k4B6CFPG005351@shell0.pdx.osdl.net>
Subject: + mm-implement-swap-prefetching-sched-batch.patch added to -mm tree
To: kernel@kolivas.org,
 mm-commits@vger.kernel.org
From: akpm@osdl.org
Date: Wed, 10 May 2006 23:09:29 -0700
X-Spam-Status: No, hits=1.088 required=5 tests=NO_REAL_NAME
X-Spam-Level: *
X-Spam-Checker-Version: SpamAssassin 2.63-osdl_revision__1.74__
X-MIMEDefang-Filter: osdl$Revision: 1.134 $
X-Scanned-By: MIMEDefang 2.36
X-DSPAM-Result: Whitelisted
X-DSPAM-Confidence: 0.9997
X-DSPAM-Probability: 0.0000
X-DSPAM-Signature: 4462d5c3152591063112884
X-DSPAM-Factors: 27,
	prefetch+static, 0.00010,
	sched+param, 0.00010,
	current+Set, 0.00010,
	sched+add, 0.00010,
	sched+add, 0.00010,
	weighted+cpu, 0.00010,
	CLASS+IDLE, 0.00010,
	devel+mm, 0.00010,
	add+above, 0.00010,
	scheduling+policy, 0.00010,
	sched+store, 0.00010,
	prevent+high, 0.00010,
	shrinker, 0.00010,
	high+load, 0.00010,
	sched+batch, 0.00010,
	sched+batch, 0.00010,
	scheduler+sys, 0.00010,
	implement+smpnice, 0.00010,
	setscheduler, 0.00010,
	lowest+if, 0.00010,
	suppressing, 0.00010,
	sched+prevent, 0.00010,
	store+weighted, 0.00010,
	sched+setscheduler, 0.00010,
	See+to, 0.00010,
	stability+of, 0.00010,
	to+lowest, 0.00010
X-Length: 4065
X-KMail-EncryptionState:  
X-KMail-SignatureState:  
X-KMail-MDN-Sent:  
X-UID: 10562
Status: RO
X-Status: OC


The patch titled

     mm: swap prefetch sched batch

has been added to the -mm tree.  Its filename is

     mm-implement-swap-prefetching-sched-batch.patch

See http://www.zip.com.au/~akpm/linux/patches/stuff/added-to-mm.txt to find
out what to do about this


From: Con Kolivas <kernel@kolivas.org>

kprefetchd is a latency insensitive ultra low priority task.  Set it to
SCHED_BATCH to obtain the benefit of this scheduling policy hint.

Signed-off-by: Con Kolivas <kernel@kolivas.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 mm/swap_prefetch.c |    3 +++
 1 files changed, 3 insertions(+)

Index: linux-2.6.16-ck11/mm/swap_prefetch.c
===================================================================
--- linux-2.6.16-ck11.orig/mm/swap_prefetch.c	2006-05-21 12:20:18.000000000 +1000
+++ linux-2.6.16-ck11/mm/swap_prefetch.c	2006-05-21 12:24:07.000000000 +1000
@@ -549,6 +549,9 @@ static enum trickle_return trickle_swap(
 
 static int kprefetchd(void *__unused)
 {
+	struct sched_param param = { .sched_priority = 0 };
+
+	sched_setscheduler(current, SCHED_BATCH, &param);
 	set_user_nice(current, 19);
 	/* Set ioprio to lowest if supported by i/o scheduler */
 	sys_ioprio_set(IOPRIO_WHO_PROCESS, 0, IOPRIO_CLASS_IDLE);
